commons/commons-util/src/main/kotlin/ebi/ac/uk/io/FileUtils.kt

```diff
@@ -152,17 +146,60 @@ internal object FileUtilsHelper {
         Files.createSymbolicLink(createParentDirectories(link, permissions), target)
     }
 
-    fun copyFolder(source: Path, target: Path, permissions: Set<PosixFilePermission>) {
+    fun copyFolder(
+        source: Path,
+        target: Path,
+        filePermissions: Set<PosixFilePermission>,
+        folderPermissions: Set<PosixFilePermission>
+    ) = applyFileVisitor(source, target, CopyFileVisitor(source, target, filePermissions, folderPermissions))
```

Comment:
I think `applyFileVisitor` is not that good as it save as like two lines of code but is completely unknow what "apply" means it is way clear IMO just call visitor.